libs/ngx-sherfire/src/lib/firestore/derivable-firestore.service.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable, NgZone } from '@angular/core';
+import { Derivable, DerivableAtom, SettableDerivable, error, lens } from '@skunkteam/sherlock';
+import { DerivableCacheOptions, derivableCache, fromEventPattern } from '@skunkteam/sherlock-utils';
+import {
+    CollectionReference,
+    DocumentData,
+    DocumentReference,
+    DocumentSnapshot,
+    FirestoreError,
+    Query,
+    QueryDocumentSnapshot,
+    QuerySnapshot,
+    onSnapshot,
+    setDoc,
+} from 'firebase/firestore';
+import { NgxSherfireModule } from '../ngx-sherfire.module';
+
+@Injectable({ providedIn: NgxSherfireModule })
+export class DerivableFirestore {
+    constructor(private readonly zone: NgZone) {}
+
+    /**
+     * Build a derivable based on a Query or CollectionReference that returns a stream of updated QuerySnapshots that contain all
+     * QueryDocumentSnapshots matching the query.
+     */
+    snapshot$<App, Db extends DocumentData>(query: Query<App, Db>): Derivable<QuerySnapshot<App, Db>>;
```

Comment:
I'd prefer `AppModel` and `DbModel` or `AppType` and `DbType` as parameter names, the names `App` and `Db` are confusing to me.